beacon-chain/state/setters.go

```diff
@@ -289,8 +299,11 @@ func (b *BeaconState) UpdateValidatorAtIndex(idx uint64, val *ethpb.Validator) e
 	v := b.state.Validators
 	if ref := b.sharedFieldReferences[validators]; ref.refs > 1 {
 		// Perform a copy since this is a shared reference and we don't want to mutate others.
-		v = b.Validators()
-
+		newVals := make([]*ethpb.Validator, len(v))
```

Comment:
As you are just copying pointers here could you not use a single `copy()` rather than a loop?

Reply:
thats much better, thanks for catching that. I have amended it